File: backend/src/main/java/edu/gsu/pantherwatch/pantherwatch/api/Faculty.java
package edu.gsu.pantherwatch.pantherwatch.api;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class Faculty {
    @JsonProperty("banner_id")
    private String bannerId;

    @JsonProperty("courseReferenceNumber")
    private String courseReferenceNumber;

    @JsonProperty("displayName")
    private String displayName;

    @JsonProperty("emailAddress")
    private String emailAddress;

    @JsonProperty("primaryIndicator")
    private String primaryIndicator;
}

File: frontend-android/app/src/main/java/edu/gsu/pantherwatch/data/model/Faculty.kt
package edu.gsu.pantherwatch.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Faculty(
    @field:Json(name = "banner_id")
    val bannerId: String?,
    
    @field:Json(name = "courseReferenceNumber")
    val courseReferenceNumber: String?,
    
    @field:Json(name = "displayName")
    val displayName: String?,
    
    @field:Json(name = "emailAddress")
    val emailAddress: String?,
    
    @field:Json(name = "primaryIndicator")
    val primaryIndicator: String?
)
